broker/src/serve_tasks.rs

```diff
@@ -87,8 +87,58 @@ async fn get_results_for_task(
     Ok((statuscode, Json(results)))
 }
 
-fn would_wait_for_elements<S>(vec: &Vec<S>, block: &HowLongToBlock) -> bool {
-    usize::from(block.wait_count.unwrap_or(0)) > vec.len()
+// GET /v1/tasks/:task_id/results/stream
+async fn get_results_for_task_stream(
+    ConnectInfo(addr): ConnectInfo<SocketAddr>,
+    State(state): State<TasksState>,
+    block: HowLongToBlock,
+    task_id: MsgId,
+    msg: MsgSigned<MsgEmpty>
+) -> Result<Sse<impl Stream<Item = Result<Event, Infallible>>>,(StatusCode, &'static str)> {
+    debug!("get_results_for_task_stream(task={}) called by {} with IP {addr}, wait={:?}", task_id.to_string(), msg.get_from(), block);
```

Comment:
Isn't a trace span or the logging middleware a better place?

Reply:
I guess that's true -- but this is just your code copied :-) Please make another PR for this (has to be changed in several similar locations). You could build on #102

Reply:
Will do 👍